activerecord/lib/active_record/test_fixtures.rb

```diff
@@ -189,7 +219,12 @@ def setup_transactional_fixtures
             if pool
               setup_shared_connection_pool
 
-              unless @fixture_connection_pools.include?(pool)
+              # Don't begin a transaction if we've already done so, or are not
+              # using transactions for this database
+              @disable_transactions =
+                (use_transactional_tests && transaction_disabled_dbnames.include?(pool.db_config.name.to_sym)) ||
+                (!use_transactional_tests && !transaction_enabled_dbnames.include?(pool.db_config.name.to_sym))
+              unless @fixture_connection_pools.include?(pool) || @disable_transactions
```

Comment:
Ah, as we need it more than once, I think let's extract a private `def transactional_tests_for_pool?(pool) = [the fetch from above]`

(With that, this won't need a variable at all -- but if this variable were remaining, it should not be an ivar.)